packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -195,47 +201,17 @@ export class WorkspaceCommandContribution implements CommandContribution {
             }
         });
         registry.registerCommand(WorkspaceCommands.NEW_FILE, this.newWorkspaceRootUriAwareCommandHandler({
-            execute: uri => this.getDirectory(uri).then(parent => {
-                if (parent) {
-                    const parentUri = new URI(parent.uri);
-                    const { fileName, fileExtension } = this.getDefaultFileConfig();
-                    const vacantChildUri = FileSystemUtils.generateUniqueResourceURI(parentUri, parent, fileName, fileExtension);
-
-                    const dialog = new SingleTextInputDialog({
-                        title: 'New File',
-                        initialValue: vacantChildUri.path.base,
-                        validate: name => this.validateFileName(name, parent, true)
-                    });
-
-                    dialog.open().then(name => {
-                        if (name) {
-                            const fileUri = parentUri.resolve(name);
-                            this.fileSystem.createFile(fileUri.toString()).then(() => {
-                                open(this.openerService, fileUri);
-                            });
-                        }
-                    });
-                }
-            })
+            execute: uri => this.getDirectory(uri).then(parent => this.createNewFile(parent))
         }));
+        registry.registerCommand(WorkspaceCommands.NEW_FILE_ROOT, {
```

Comment:
Why do we need it? Should not we have one `NEW_FILE` command which work based on a selection in the navigator?

i.e. if there is not a selection only then a file should be created in the first root always, otherwise in the selected folder?

Reply:
I'll support this but updating the command-handler passed.
I thought of doing this as well but I wanted to try and limit the number of commands which had to be adjusted/updated. It looks like there is in fact no way around it but updating the commands to fallback to the root.